linc_convert/utils/zarr_config.py

```diff
@@ -96,33 +96,64 @@ class ZarrConfig:
     no_pyramid_axis: Literal["x", "y", "z"] | None = None
     levels: int = -1
     ome_version: Literal["0.4", "0.5"] = "0.4"
-    nii: bool = False
-    max_load: int = 512
-    overwrite: bool = True
+    overwrite: bool = False
     driver: DriverLike = "zarr-python"
 
     def __post_init__(self) -> None:
         """
         Perform post-initialization checks and adjustments.
 
-        - Automatically enable NIfTI mode if the output path ends with ".nii.zarr".
         - Ensure that sharding options (shard, shard_channels, shard_time) are only
           used when zarr_version == 3; otherwise raise NotImplementedError.
         """
-        if self.out:
-            self.nii |= str(self.out).endswith(".nii.zarr")
         if self.zarr_version == 2:
             if self.shard or self.shard_channels or self.shard_time:
-                raise NotImplementedError("Shard is not supported for Zarr 2.")
+                raise ValueError("Shard is not supported for Zarr 2.")
 
-    def set_default_name(self, name: str) -> None:
+
+@Parameter(name="*")
+@dataclass
+class GeneralConfig:
+    """
+    General configuration for the conversion process.
+
+    Parameters
+    ----------
+    out : str | None
+        Output path for the converted data.
+    max_load : int
+        Maximum number of items to load into memory at once.
+    """
+
+    out: Annotated[str, Parameter(name=["--out", "-o"])] = None
+    max_load: int = 1024
+    log_level: Literal["debug", "info", "warning", "error", "critical"] = "info"
+    verbose: Annotated[bool, Parameter(name=["--verbose", "-v"])] = False
+
+    def __post_init__(self) -> None:
+        """
+        Perform post-initialization checks and adjustments.
+
+        - Ensure that max_load is a positive integer; otherwise raise ValueError.
+        - If verbose is True, set log_level to "debug".
+        """
+        if self.max_load <= 0:
```

Comment:
To check if it is an integer.

```suggestion
        if not isinstance(self.max_load, int) or self.max_load <= 0:
```